subsys/shell/shell_log_backend.c

```diff
@@ -18,10 +18,25 @@ int shell_log_backend_output_func(u8_t *data, size_t length, void *ctx)
 void shell_log_backend_enable(const struct shell_log_backend *backend,
 			      void *ctx, u32_t init_log_level)
 {
-	log_backend_enable(backend->backend, ctx, init_log_level);
-	log_output_ctx_set(backend->log_output, ctx);
-	backend->control_block->cnt = 0;
-	backend->control_block->state = SHELL_LOG_BACKEND_ENABLED;
+	int err = 0;
+
+	if (IS_ENABLED(CONFIG_LOG_INPLACE_PROCESS)) {
+		const struct shell *shell;
+
+		shell = (const struct shell *)ctx;
+
+		/* Reenable transport in blocking mode */
+		err = shell->iface->api->enable(shell->iface, true);
```

Comment:
Could we have only one call to "enable" during startup?

Reply:
no, `enable` enables transport in given mode (blocking or nonblocking). Shell is by default started as asynchronous but can be reconfigured to blocking by the logger.